src/core/meshcop/dataset.hpp

```diff
@@ -648,47 +648,94 @@ class Dataset
     }
 
     /**
-     * Returns a pointer to the TLV with a given type.
+     * Searches for a given TLV type in the Dataset.
      *
-     * @param[in] aType  A TLV type.
+     * @param[in] aType  The TLV type to find.
      *
-     * @returns A pointer to the TLV or `nullptr` if none is found.
+     * @returns A pointer to the TLV or `nullptr` if not found.
      *
      */
-    Tlv *GetTlv(Tlv::Type aType) { return AsNonConst(AsConst(this)->GetTlv(aType)); }
+    Tlv *FindTlv(Tlv::Type aType) { return AsNonConst(AsConst(this)->FindTlv(aType)); }
 
     /**
-     * Returns a pointer to the TLV with a given type.
+     * Searches for a given TLV type in the Dataset.
      *
-     * @param[in] aType  The TLV type.
+     * @param[in] aType  The TLV type to find.
      *
-     * @returns A pointer to the TLV or `nullptr` if none is found.
+     * @returns A pointer to the TLV or `nullptr` if not found.
      *
      */
-    const Tlv *GetTlv(Tlv::Type aType) const;
+    const Tlv *FindTlv(Tlv::Type aType) const;
 
     /**
-     * Returns a pointer to the TLV with a given template type `TlvType`
+     * Writes a TLV to the Dataset.
      *
-     * @returns A pointer to the TLV or `nullptr` if none is found.
+     * If the specified TLV type already exists, it will be replaced. Otherwise, the TLV will be appended.
+     *
+     * @param[in] aTlv A reference to the TLV.
+     *
+     * @return kErrorNone Success: The TLV was successfully updated.
+     * @return kErrorNoBufs Failure: Insufficient buffer space prevented the TLV from being added.
```

Comment:
```suggestion
     * @return kErrorNone    The TLV was successfully updated.
     * @return kErrorNoBufs  Insufficient buffer space prevented the TLV from being added.
```

Reply:
Ah! Good find. 
Not sure where these came from? Also will change them to `@retval`.